src/main/java/stream/flarebot/flarebot/commands/general/SelfAssignCommand.java

```diff
@@ -43,16 +44,17 @@ public void onCommand(User sender, GuildWrapper guild, TextChannel channel, Mess
                             5000, channel);
                 }
             } else if (args[0].equalsIgnoreCase("list")) {
-                StringBuilder sb = new StringBuilder();
                 if (FlareBotManager.getInstance().getSelfAssignRoles(channel.getGuild().getId()).isEmpty()) {
                     MessageUtils.sendAutoDeletedMessage(MessageUtils.getEmbed(sender).setColor(Color.RED).setDescription("There are no self-assignable roles!").build(), 5000, channel);
                     return;
                 }
-                sb.append("**Self assignable roles**").append("\n```\n");
-                FlareBotManager.getInstance().getSelfAssignRoles(channel.getGuild().getId()).forEach(role -> sb.append(channel.getGuild().getRoleById(role).getName()).append(" (")
-                        .append(role).append(")\n"));
-                sb.append("```");
-                channel.sendMessage(sb.toString()).queue();
+                String base = "**Self assignable roles**\n```\n";
+                base += FlareBotManager.getInstance().getSelfAssignRoles(channel.getGuild().getId())
+                        .stream().filter(roleId -> channel.getGuild().getRoleById(roleId) != null)
+                        .map(roleId -> channel.getGuild().getRoleById(roleId).getName() + " (" + roleId + ")")
```

Comment:
How would you suggest me doing that? Because the event handler should do that this is just to handle edge cases